library/systemd/src/lib/yast2/compound_service.rb

```diff
@@ -0,0 +1,202 @@
+  class CompoungService
+    include Yast::Logger
+
+    # managed services
+    # @return [Yast2::ServiceConfiguration]
+    attr_reader :services
+
+    # creates new configuration that holds state for given services
+    # @param services<Yast::SystemdServiceClass::Service> services to configure
+    # @param reload<true,false> if use reload instead of restart action. If
+    #   service does not support reload or does not run, then restart is used.
+    #
+    # @example three services
+    #   config = Yast2::ServiceConfiguration.new(s1, s2, s3)
+    def initialize(*services)
+      if services.any? { |s| !s.is_a?(Yast2::SystemdService) }
+        raise ArgumentError, "Services can be only System Service - #{services.inspect}"
+      end
+
+      @services = services
+    end
+
+    # writes services new status
+    # @raise <Yast::SystemctlError> when set service to target state failed
+    def save(ignore_status: false)
+      services.each { |s| s.save(ignore_status: ignore_status) }
+      reset # to be consistent with underlaying services
+    end
+
+    # returns current running status.
+    # @return [true, false, :inconsistent] current status of
+    #   services:
+    #
+    #   - `true` when all services are active
+    #   - `false` when all services are inactive
+    #   - `:inconsistent` when part of services is active and part not.
+    def current_active?
+      return true if services.all?(&:current_active?)
+      return false if services.none?(&:current_active?)
+      :inconsistent
+    end
+
+    # returns true if any service supports reload
+    def support_reload?
+      services.any?(&:support_reload?)
+    end
+
+    def start_modes
+      @start_modes ||= services.map(&:start_modes).reduce(:+).uniq
+    end
+
+    def search_term
+      services.map(&:search_term).reduce(:+).uniq
+    end
+
+    # returns currently set target action. If it is not yet set it returns
+    # `:nothing`.
+    # @return [:start, :stop, :restart, :nothing] action for all services:
+    #
+    #   - `:start` to start all services. If service is already active, do nothing.
+    #     if services has socket it starts socket instead of service.
+    #   - `:stop`  to stop all services. If service already is inactive, do nothing.
+    #   - `:restart` restart all services. If service is inactive, it is started.
+    #   - `:reload` reload all services that support it and restart that does not
+    #       support it. If service is inactive, it is started.
+    #   - `nil` do not touch anything.
+    def action
+      # TODO: check for inconsistencies?
+      services.first.action
+    end
+
+    ACTIONS = [:start, :stop, :reload, :restart, :nothing].freeze
+    # sets target action for services
+    # @param action[:start, :stop, :restart, :reload, :nothing] for possible values and
+    #  its explanation please see return value of {target_action}
+    def action=(action)
```

Comment:
Actually needed? We have separate actions `#start`, `#stop`, etc.

Reply:
no, not needed, I forgot to remove it.